appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/NestedAppClientDeployerHelper.java

```diff
@@ -352,14 +352,14 @@ private void processDependencies() throws IOException {
 
         URI appClientURIWithinEAR = URI.create(appClientDesc().getModuleDescriptor().getArchiveUri());
         final Artifact appClientJARArtifact = newArtifact(appClientURIWithinEAR);
-
-        /*
-         * Processing the client artifact will recursively process any artifacts
-         * on which it depends plus the transitive closure thereof.
-         */
-        appClientJARArtifact.processArtifact(dependencyURIsProcessed,
-                clientLevelDownloads(), earLevelDownloads());
-
+        if (appClientJARArtifact != null) {
```

Comment:
In other places `newArtifact` was used (which can return null) a null check was performed so I opted for the same.